### PR TITLE
Add integer support for eltwise ops

### PR DESCRIPTION
**Pull request by mouliraj-mcw**
### Ticket
Link to Github Issue #13374

### What's changed

-  Updated the comparison ops to return datatype as integer.

Ops Added Int Support:

- ELEMWISE_UNARY_EQ
- ELEMWISE_UNARY_GE
- ELEMWISE_UNARY_GT
- ELEMWISE_UNARY_LE
- ELEMWISE_UNARY_LT
- ELEMWISE_UNARY_NE
- ELEMWISE_BINARY_GE
- ELEMWISE_BINARY_GT
- ELEMWISE_BINARY_EQ
- ELEMWISE_BINARY_LE
- ELEMWISE_BINARY_LOGICALAND
- ELEMWISE_BINARY_LOGICALOR
- ELEMWISE_BINARY_LOGICALXOR
- ELEMWISE_BINARY_LT
- ELEMWISE_BINARY_NE

### Checklist
- [ ] [All Post commit CI](https://github.com/tenstorrent/tt-metal/actions/runs/11796923243)

### CODE REVIEW

**Review by umadevimcw — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/eltwise/test_binary_comp_init.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import pytest
+import ttnn
+from tests.ttnn.unit_tests.operations.eltwise.backward.utility_funcs import data_gen_with_range
+from models.utility_functions import is_grayskull, skip_for_blackhole
+
+
+@skip_for_blackhole("Mismatching on BH, see #12349")
+@pytest.mark.parametrize(
+    "input_shapes",
+    ((torch.Size([1, 1, 32, 32])),),
```
Comment (KalaivaniMCW) [inline tests/ttnn/unit_tests/operations/eltwise/test_binary_comp_init.py:15]: add a few more shapes from rank 2 to 5

File: tests/ttnn/unit_tests/operations/eltwise/test_binary_comp_init.py
Comment (KalaivaniMCW) [inline tests/ttnn/unit_tests/operations/eltwise/test_binary_comp_init.py:1]: please add test cases passing optional output tensor for both variants here

**Comment by KalaivaniMCW** [comment]
@mouliraj-mcw Please list the ops that are being handled in this PR in the description.

**Review by VirdhatchaniKN — APPROVED** [review]

**Review by mcw-anasuya — APPROVED** [review]

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by eyonland — APPROVED** [review]

**Pull request merged**